setuptools/command/egg_info.py

```diff
@@ -618,7 +618,15 @@ def _safe_data_files(self, build_py):
         Therefore, avoid triggering any attempt of
         analyzing/building the manifest again.
         """
-        return build_py.get_data_files_without_manifest()
+        if hasattr(build_py, 'get_data_files_without_manifest'):
+            return build_py.get_data_files_without_manifest()
+
+        log.warn(
```

Comment:
I'm unsure whether to invoke log.warn here or issue a DeprecationWarning. The latter can be controlled by warnings filters. In any case, I quite like the approach taken here to direct the users away from using distutils, which _is_ deprecated.

Reply:
I wasn't sure myself 😅 

If you prefer a DeprecationWarning, I can work in a change over the weekend.